tools/src/block-gen/Main.hs

```diff
@@ -0,0 +1,83 @@
+module Main where
+
+import           Universum
+
+import qualified Data.HashMap.Strict as HM
+import qualified Data.Map            as M
+import           Formatting          (build, sformat, (%))
+import           Mockable            (runProduction)
+import           System.Directory    (doesDirectoryExist)
+
+import           Pos.Core            (StakeDistribution (..), addressHash,
+                                      genesisDevKeyPairs, genesisProdAddrDistribution,
+                                      isDevelopment, makePubKeyAddress, mkCoin)
+import           Pos.Crypto          (SecretKey, toPublic)
+import           Pos.DB              (closeNodeDBs, openNodeDBs)
+import           Pos.Generator.Block (AllSecrets (..), BlockGenParams (..), genBlocks)
+import           Pos.Genesis         (devAddrDistr, genesisUtxo)
+import           Pos.Txp.Core        (TxOut (..), TxOutAux (..))
+import           Pos.Txp.Toil        (GenesisUtxo (..), Utxo)
+import           Pos.Util.UserSecret (peekUserSecret, usPrimKey)
+
+import           Context             (initTBlockGenMode)
+import           Error               (TBlockGenError (..))
+import           Options             (BlockGenOptions (..), getBlockGenOptions)
+
+main :: IO ()
+main = flip catch catchEx $ do
+    BlockGenOptions{..} <- getBlockGenOptions
+    when bgoAppend $ checkExistence bgoPath
+    secretsMap <- case bgoNodes of
+        Left bgoNodesN -> do
+            let keys = take (fromIntegral bgoNodesN) genesisDevKeyPairs
+            let secretsMap = HM.fromList $ map (first addressHash) keys
+            pure secretsMap
+        Right bgoSecretFiles -> do
+            when (null bgoSecretFiles) $
+                throwM NoOneSecrets
+            secrets <- runProduction $ mapM parseSecret bgoSecretFiles
+            let secretsMap = HM.fromList $
+                                map (first (addressHash . toPublic) . join (,)) secrets
+            pure secretsMap
+
+    let nodes = length secretsMap
+    let flatDistr = FlatStakes (fromIntegral nodes) (mkCoin $ fromIntegral nodes)
```

Comment:
I am not sure it's ok to hardcode `FlatStakes`. Probably benchmarkers use different distribution and they may need this tool. At least you should create an issue to support other distributions.